packages/dx-react-chart/src/plugins/zoom-pan.tsx

```diff
@@ -173,34 +240,28 @@ class ZoomAndPanBase extends React.PureComponent<ZoomAndPanProps, ZoomAndPanStat
       <Getter name="ranges" computed={getAdjustedLayout} />
         <Template name="root">
           <TemplateConnector>
-            {({ scales }) =>
+            {({ scales, rootRef }) =>
             <DragDropProvider>
               <DropTarget
                 onOver={({ _, clientOffset }) => this.handleMouseMove(scales, clientOffset)}
                 onDrop={() => this.handleMouseUp(scales)}
               >
                 <DragSource payload={null}>
-                  <TemplatePlaceholder />
+                  <ZoomPanProvider
```

Comment:
It does not need to render anything (its purpose is to access *mount* and *unmount* stages). So it should be moved out of *DragDropProvider* (hence be independent of provider's renders).

```jsx
<ZoomPanProvider ... />
<DragDropProvider ... />
```

Reply:
React classes should render something, in this case it is `TemplatePlaceholder`. If `ZoomPanProvider` renders `null`, zooming wouldn't be work.